.github/workflows/container-image.yml

```diff
@@ -101,25 +107,85 @@ jobs:
+      -
+        name: Create SBOM file of build
+        if: ${{ inputs.generate-sbom == true }}
+        shell: bash
+        run: |
+          bom generate -n https://kubewarden.io/kubewarden.spdx \
+            --format json \
+            . > kubewarden-controller-sboms/kubewarden-controller-sbom.spdx
+      -
+        name: Sign SBOM files of build
+        if: ${{ inputs.generate-sbom == true }}
+        run: |
+          cosign sign-blob --output-certificate kubewarden-controller-sboms/kubewarden-controller-sbom.spdx.cert \
+            --output-signature kubewarden-controller-sboms/kubewarden-controller-sbom.spdx.sig \
+            kubewarden-controller-sboms/kubewarden-controller-sbom.spdx
+        env:
+          COSIGN_EXPERIMENTAL: 1
+      -
+        name: Upload SBOM files of build
+        uses: actions/upload-artifact@v2
+        if: ${{ inputs.generate-sbom == true }}
+        with:
+          name: kubewarden-controller-sboms
+          path: |
+            kubewarden-controller-sboms/
+      -
+        name: Create directory to store SBOM files of container image
         if: ${{ inputs.generate-sbom == true }}
         shell: bash
         run: |
-          bom generate -n https://kubewarden.io/kubewarden.spdx -o kubewarden-controller.spdx .
+          mkdir kubewarden-controller-container-image-sboms
       -
-        name: Attach SBOM file in the container image
+        name: Create SBOM file of container image
+        if: ${{ inputs.generate-sbom == true }}
+        shell: bash
+        run: |
+          bom generate -n https://kubewarden.io/kubewarden.spdx \
+            --format json \
+            --image "ghcr.io/${{github.repository_owner}}/kubewarden-controller@${{ steps.build-image.outputs.digest }}" \
+            . > kubewarden-controller-container-image-sboms/kubewarden-controller-container-image-sbom.spdx
+      -
+        name: Sign SBOM files of container image
+        if: ${{ inputs.generate-sbom == true }}
+        run: |
+          cosign sign-blob --output-certificate kubewarden-controller-container-image-sboms/kubewarden-controller-container-image-sbom.spdx.cert \
+            --output-signature kubewarden-controller-container-image-sboms/kubewarden-controller-container-image-sbom.spdx.sig \
+            kubewarden-controller-container-image-sboms/kubewarden-controller-container-image-sbom.spdx
+        env:
+          COSIGN_EXPERIMENTAL: 1
+      -
+        name: Attach SBOM file of build in the container image
         if: ${{ inputs.generate-sbom == true }}
         shell: bash
         run: |
           set -e
-          cosign attach sbom --sbom kubewarden-controller.spdx "ghcr.io/${{github.repository_owner}}/kubewarden-controller@${{ steps.build-image.outputs.digest }}"
+          cosign attach sbom --sbom kubewarden-controller-container-image-sboms/kubewarden-controller.spdx \
+            "ghcr.io/${{github.repository_owner}}/kubewarden-controller@${{ steps.build-image.outputs.digest }}"
```

Comment:
we don't sign the sbom we attach. We should probably sign it after this or use in-toto attestations
more info https://docs.sigstore.dev/cosign/other_types/#sboms-software-bill-of-materials

Reply:
I saw we are signing it in a different step, sorry about the noise